File: web_admin/src/domain/tags/tagColors.test.ts
import { describe, expect, it } from 'vitest';
import { TAG_COLORS, normalizeTagColor, tagChipStyle } from './tagColors';

describe('tagColors', () => {
  it('exposes the eight canonical tokens', () => {
    expect(TAG_COLORS).toEqual([
      'gray', 'red', 'amber', 'green', 'teal', 'blue', 'purple', 'pink',
    ]);
  });
  it('normalizes unknown / missing values to gray', () => {
    expect(normalizeTagColor('green')).toBe('green');
    expect(normalizeTagColor('chartreuse')).toBe('gray');
    expect(normalizeTagColor(undefined)).toBe('gray');
    expect(normalizeTagColor(null)).toBe('gray');
    expect(normalizeTagColor(42)).toBe('gray');
  });
  it('every token has a chip style', () => {
    for (const c of TAG_COLORS) {
      const s = tagChipStyle(c);
      expect(s.bg).toMatch(/^#/);
      expect(s.fg).toMatch(/^#/);
    }
  });
});

File: test/core/constants/tag_colors_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:maki_mobile_pos/core/constants/tag_colors.dart';

void main() {
  test('eight canonical tokens, in lockstep with the web list', () {
    expect(TagColors.tokens, [
      'gray', 'red', 'amber', 'green', 'teal', 'blue', 'purple', 'pink',
    ]);
  });

  test('normalize falls back to gray', () {
    expect(TagColors.normalize('green'), 'green');
    expect(TagColors.normalize('neon'), 'gray');
    expect(TagColors.normalize(null), 'gray');
  });

  test('styleFor resolves every token in both brightnesses', () {
    for (final token in TagColors.tokens) {
      expect(TagColors.styleFor(token, false).bg, isNotNull);
      expect(TagColors.styleFor(token, true).fg, isNotNull);
    }
    // Unknown token renders as gray, never throws.
    expect(TagColors.styleFor('neon', false).bg, TagColors.styleFor('gray', false).bg);
  });
}
